app/hotels/src/map/MapView.js

```diff
@@ -0,0 +1,133 @@
+// @flow
+
+import * as React from 'react';
+import { StyleSheet, View } from 'react-native';
+import { createFragmentContainer, graphql } from 'react-relay';
+import idx from 'idx';
+
+import Map from './Map';
+import HotelSwipeList from './HotelSwipeList';
+import type { MapView as MapViewData } from './__generated__/MapView.graphql';
+
+type Props_ = {|
+  data: MapViewData,
+|};
+
+type Props = Props & $Shape<Props_>;
+
+type State = {|
+  selectedHotelIndex: number,
+|};
+
+const styles = StyleSheet.create({
+  container: {
+    ...StyleSheet.absoluteFillObject,
+    justifyContent: 'flex-end',
+    alignItems: 'center',
+  },
+});
+
+class MapView extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      selectedHotelIndex: 0,
+    };
+  }
+
+  componentWillReceiveProps = (nextProps: Props) => {
+    if (this.props.data !== nextProps.data) {
+      // TODO keep correct index if node with the same id is still present
+      this.setState({ selectedHotelIndex: 0 });
+    }
+  };
+
+  getHotels = () => {
+    return idx(this.props, _ => _.data.edges) || [];
+  };
+
+  selectMarker = (selectedHotelIndex: number) => {
+    if (this.state.selectedHotelIndex === selectedHotelIndex) {
+      return;
+    }
+
+    this.setState({ selectedHotelIndex });
+  };
+
+  onSelectMarker = (hotelId: string) => {
+    const hotels = this.getHotels();
+    const index = hotels.findIndex(
+      hotel => idx(hotel, _ => _.node.id) === hotelId,
+    );
+
+    if (Number.isFinite(index)) {
```

Comment:
This condition will be always true, it should be `if (index !== -1)` (when `findIndex` doesn't find anything, it returns -1 and `Number.isFinite(-1) === true`).

Reply:
good spot, thanks!